src/types/sounds.ts

```diff
@@ -0,0 +1,68 @@
+import * as Yup from "yup";
+import { SpriteMap } from "use-sound/dist/types";
+
+import { WithId } from "utils/id";
+
+export interface SoundConfig {
+  url: string;
+  sprites?: SpriteMap;
+}
+
+export interface SoundConfigReference {
+  soundId: string;
+  spriteName?: string;
+}
+
+export type SoundConfigMap = Partial<Record<string, WithId<SoundConfig>>>;
+
+/**
+ * SpriteMap validation schema.
+ *
+ * @see SpriteMap
+ */
+export const SpriteMapSchema = Yup.lazy<{} | undefined>((data = {}) => {
+  const lazyObjectShape: Record<
+    string,
+    Yup.ArraySchema<number>
+  > = Object.entries(data).reduce(
+    (acc, [key]) => ({
+      ...acc,
+      [key]: Yup.array().of(Yup.number().required()).min(2).max(2).required(),
+    }),
+    {}
+  );
+
+  return Yup.object().shape(lazyObjectShape).noUnknown();
+});
+
+/**
+ * SoundConfig validation schema.
+ *
+ * @see SoundConfig
+ */
+export const SoundConfigSchema: Yup.ObjectSchema<SoundConfig> = Yup.object()
+  .shape<SoundConfig>({
+    url: Yup.string().required(),
+    sprites: SpriteMapSchema,
+  })
+  .noUnknown()
+  .required();
```

Comment:
[note] The schema used for proactively validating data when it is retrieved from firestore to ensure that it matches what we expect (aka: no one can break it by hacking the database without us knowing, thus avoiding weird/hard to debug/find bugs)